app/parse.py

```diff
@@ -15,8 +28,119 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = tuple(field.name for field in fields(Product))
+
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def scrape_one_product(product: Tag) -> Product:
+    return Product(
+        title=product.select_one(".title")["title"],
+        description=product.select_one(".description").text.strip().replace(
+            "\xa0", " "
+        ),
+        price=float(product.select_one(".price").text.lstrip("$")),
+        rating=len(product.select(".ws-icon-star")),
+        num_of_reviews=int(
+            product.select_one(".review-count").text.split(" ")[0]
+        ),
+    )
+
+
+def get_soup(url: str, session: requests.Session) -> BeautifulSoup:
+    response = session.get(url)
+    return BeautifulSoup(response.content, "html.parser")
+
+
+def parse_one_page(url: str) -> [Product]:
```

Comment:
The return type hint for `parse_one_page` should be `List[Product]` instead of `[Product]`. This will require importing `List` from the `typing` module.